doc/man7/provider-rand.pod

```diff
@@ -144,8 +145,13 @@ OSSL_FUNC_rand_enable_locking() allows locking to be turned on for a DRBG and al
 its parent DRBGs.  From this call onwards, the DRBG can be used in a thread
 safe manner.
 
-OSSL_FUNC_rand_lock() is used to lock a DRBG.  Once locked, exclusive access
-is guaranteed.
+OSSL_FUNC_rand_lock() and OSSL_FUNC_rand_lock_ex() are used to lock a DRBG.
+Where possible it is preferred that a DRBG implement OSSL_FUNC_rand_lock_ex()
+in preference to OSSL_FUNC_rand_lock(). Once locked, exclusive access
+is guaranteed. OSSL_FUNC_rand_lock_ex() takes an additional "read" parameter
```

Comment:
We should implement both - remember we need forwards and backwards compatibility.

Reply:
Surely there is a difference between what we implement ourselves and what we require others to implement? This PR implements both for our own DRBGs to obtain that forward and backwards compat. I don't think we need to *require* other provider authors to implement both. Although we probably should document what happens if they don't implement the older version. To get the text right for that though I think we need to decide which branches this change is going into (and there is an OTC hold for that)

Reply:
We shouldn't insist other providers implement both, but we have to support either being implemented.
We should, however, implement both.